src/api/request-validations.ts

```diff
@@ -39,3 +41,21 @@ export const validateBodyEquals = (request: RequestExt, expectedBody: unknown |
 const isBodyEmpty = (body: string | undefined | null) => {
   return body == null || body === "";
 };
+
+const coerceDateXml = (xml: string): string => {
+  return xml.replace(/(\d\d\d\d-\d\d-\d\d[Tt]\d\d:\d\d:\d\d\.\d\d\d)\d{0,4}([Zz]|[+-]00:00)/g, "$1Z");
+};
+
+// Check whether the XML request body is valid
+
+export const validateXMLBodyEquals = (request: RequestExt, expectedBody: string): void => {
+  const rawBody = request.body;
+  const actualBody = coerceDateXml(rawBody);
+  parseString(actualBody, (err, actualParsedBody) => {
```

Comment:
This is going to be async with the callback, try using `parseStringPromise` instead with `await` and making this function async and cascading upstream
https://www.npmjs.com/package/xml2js#promise-usage